src/main/java/org/primeframework/mvc/content/json/BaseJacksonContentHandler.java

```diff
@@ -109,6 +110,9 @@ public void handle() throws IOException {
         if (logger.isDebugEnabled()) {
           String body = new String(request.getBodyBytes(), 0, contentLength.intValue());
           logger.debug("Request: ({} {}) {}", request.getMethod(), request.getPath(), body);
+
+          // Replace the input stream, in case anything downstream wants to use it
+          request.setInputStream(new ByteArrayInputStream(body.getBytes(request.getCharacterEncoding())));
```

Comment:
This seems reasonable. 

The alternative would be for the sub-classes to account for this, or to move the debug output into the sub-classes so they can account for the change in behavior. 

@voidmain any opinion before we move forward?

Reply:
This works fine for now. The InputStream is the main source of the body bytes, so replacing it with another stream is fine.